pkg/config/validation_test.go

```diff
@@ -6,26 +22,28 @@ import (
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/ptr"
 
 	configapi "sigs.k8s.io/kueue/apis/config/v1beta1"
 )
 
-func TestValidate(t *testing.T) {
+func TestValidateQueueVisibility(t *testing.T) {
```

Comment:
I think it is generally better to test the exported functions. Thus, I would suggest to keep TestValidate, and pass the entire config for the new test cases. Any reason why not to do this?

Reply:
We can do the black box testing, for sure. It will just require some defaulting for all the cases since exported function will validate multiple parts of the configuration.

Reply:
Yeah, this is what we do for validation of k8s jobs: https://github.com/kubernetes/kubernetes/blob/d393d4e1514711e89b310d4578214c8c383e015c/pkg/apis/batch/validation/validation_test.go#L90. The good thing is that it does not make an extra effort to refactor internal functions.